monasca_events_api/v2/transforms.py

```diff
@@ -163,28 +177,22 @@ def _create_transform_response(self, transform_id, transform):
                     'specification': specification, 'enabled': enabled}
         return json.dumps(response)
 
+    @resource.resource_try_catch_block
     def _list_transforms(self, tenant_id, limit, offset, uri):
-        try:
-            transforms = self._transforms_repo.list_transforms(tenant_id,
-                                                               limit, offset)
-            transforms = helpers.paginate(transforms, uri)
-            return transforms
-        except repository_exceptions.RepositoryException as ex:
-            LOG.error(ex)
-            raise falcon.HTTPInternalServerError('Service unavailable',
-                                                 ex.message)
 
+        transforms = self._transforms_repo.list_transforms(tenant_id,
+                                                           limit, offset)
+        transforms = helpers.paginate(transforms, uri)
+        return transforms
+
+
+    @resource.resource_try_catch_block
     def _list_transform(self, tenant_id, transform_id, uri):
-        try:
-            transform = self._transforms_repo.list_transform(tenant_id,
-                                                             transform_id)[0]
-            transform['specification'] = yaml.safe_dump(
-                transform['specification'])
-            return transform
-        except repository_exceptions.RepositoryException as ex:
-            LOG.error(ex)
-            raise falcon.HTTPInternalServerError('Service unavailable',
-                                                 ex.message)
+        transform = self._transforms_repo.list_transform(tenant_id,
+                                                         transform_id)[0]
+        transform['specification'] = yaml.safe_dump(
+            transform['specification'])
+        return transform
 
     def _delete_transform(self, tenant_id, transform_id):
         try:
```

Comment:
I think we need to figure out where we want to use the resource_try_catch_block decorator in transforms.py.  At the least it looks like _create_transform needs one.

I wonder if we want to bother decorating low level functions like this though.  Why not just decorate the on_post, on_get, on_delete, and on_patch methods?  We can add the proper error handling to deal with bad input and errors writing to kafka to the resource_try_catch_block, and possibly rename it, so we can deal with those sorts of errors in one place instead of in each HTTP handler class.